src/pudl/extract/xbrl.py

```diff
@@ -0,0 +1,166 @@
+"""Generic extractor for all FERC XBRL data."""
+import io
+import json
+import time
+from datetime import datetime
+
+import sqlalchemy as sa
+from ferc_xbrl_extractor import xbrl
+from ferc_xbrl_extractor.instance import InstanceBuilder
+
+import pudl
+from pudl.helpers import get_logger
+from pudl.settings import FercGenericXbrlToSqliteSettings, FercToSqliteSettings
+from pudl.workspace.datastore import Datastore
+
+logger = get_logger(__name__)
+
+XBRL_FORM_NUMBERS = [1, 2, 6, 60, 714]
```

Comment:
Maybe constrain with an Enum?

Reply:
Good call. Switched to an enum, which also clarifies allowable values in arguments

Reply:
I think I found some cut and paste errors related to this switch, commented in another review.